contracts/0.4.24/Lido.sol

```diff
@@ -344,101 +323,45 @@ contract Lido is ILido, StETH, AragonApp {
-        uint16 _insuranceFeeBasisPoints,
-        uint16 _operatorsFeeBasisPoints
-    )
-        external
-    {
+    function setMaxFee(uint16 _maximumFeeBasisPoints) external {
         _auth(MANAGE_FEE);
 
-        require(
-            TOTAL_BASIS_POINTS == uint256(_treasuryFeeBasisPoints)
-            .add(uint256(_insuranceFeeBasisPoints))
-            .add(uint256(_operatorsFeeBasisPoints)),
-            "FEES_DONT_ADD_UP"
-        );
-
-        _setBPValue(TREASURY_FEE_POSITION, _treasuryFeeBasisPoints);
-        _setBPValue(INSURANCE_FEE_POSITION, _insuranceFeeBasisPoints);
-        _setBPValue(NODE_OPERATORS_FEE_POSITION, _operatorsFeeBasisPoints);
-
-        emit FeeDistributionSet(_treasuryFeeBasisPoints, _insuranceFeeBasisPoints, _operatorsFeeBasisPoints);
+        _setBPValue(MAX_FEE_POSITION, _maximumFeeBasisPoints);
+        emit MaxFeeSet(_maximumFeeBasisPoints);
     }
 
     /**
-    * @notice Set Lido protocol contracts (oracle, treasury, insurance fund).
-    *
-    * @dev Oracle contract specified here is allowed to make
-    * periodical updates of beacon stats
-    * by calling pushBeacon. Treasury contract specified here is used
-    * to accumulate the protocol treasury fee. Insurance fund contract
-    * specified here is used to accumulate the protocol insurance fee.
-    *
-    * @param _oracle oracle contract
-    * @param _treasury treasury contract
-    * @param _insuranceFund insurance fund contract
-    */
-    function setProtocolContracts(
-        address _oracle,
-        address _treasury,
-        address _insuranceFund
-    ) external {
+     * @notice Set Lido protocol contracts (oracle, treasury).
+     *
+     * @dev Oracle contract specified here is allowed to make
+     * periodical updates of beacon stats
+     * by calling pushBeacon. Treasury contract specified here is used
+     * to accumulate the protocol treasury fee.
+     *
+     * @param _oracle oracle contract
+     * @param _treasury treasury contract
+     */
+    function setProtocolContracts(address _oracle, address _treasury) external {
         _auth(MANAGE_PROTOCOL_CONTRACTS_ROLE);
 
-        _setProtocolContracts(_oracle, _treasury, _insuranceFund);
-    }
-
-    /**
-    * @notice Set credentials to withdraw ETH on ETH 2.0 side after the phase 2 is launched to `_withdrawalCredentials`
-    * @dev Note that setWithdrawalCredentials discards all unused signing keys as the signatures are invalidated.
-    * @param _withdrawalCredentials withdrawal credentials field as defined in the Ethereum PoS consensus specs
-    */
-    function setWithdrawalCredentials(bytes32 _withdrawalCredentials) external {
-        _auth(MANAGE_WITHDRAWAL_KEY);
-
-        WITHDRAWAL_CREDENTIALS_POSITION.setStorageBytes32(_withdrawalCredentials);
-        getOperators().trimUnusedKeys();
-
-        emit WithdrawalCredentialsSet(_withdrawalCredentials);
+        _setProtocolContracts(_oracle, _treasury);
+        emit ProtocolContactsSet(_oracle, _treasury);
```

Comment:
duplicated event `ProtocolContactsSet` let's fix

Reply:
let's remove here and keep inside internal func

Reply:
fixed